### PR TITLE
Enable gtest for the project

### PR DESCRIPTION
**Pull request by dimitry-**
Adds dependency and a simple test.

### CODE REVIEW

File: OrbitCore/CMakeLists.txt [hunk 1]
```diff
@@ -198,3 +198,11 @@ target_compile_definitions(OrbitCore PUBLIC UNICODE)
 target_compile_definitions(OrbitCore PUBLIC _UNICODE)
 
 target_compile_features(OrbitCore PUBLIC cxx_std_11)
+
+add_executable(OrbitCoreTests)
+
+target_sources(OrbitCoreTests
+  PRIVATE RingBufferTests.cpp)
+
+target_link_libraries(
```
Comment (beckerhe) [inline OrbitCore/CMakeLists.txt]: The OrbitCoreTests target should probably depend on the OrbitCore target.

If ringbuffer.h wasn't header-only and didn't happen to be in the same include directory, you would get a compile or linker error.

Comment (dimitry-) [reply OrbitCore/CMakeLists.txt]: done

**Comment by pierricgimmig** [comment]
Hi @dimitry- , that's great!  Thanks.  I'll wait until @beckerhe cmake change is merged before merging this PR.

**Comment by pierricgimmig** [comment]
@dimitry- , please merge latest commit of headless branch.

**Pull request merged**